### PR TITLE
INTEL MKL: Fixing api_compatibility_test

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR fixes the api_compatibility_test by hiding MKL ops.

### CODE REVIEW

File: tensorflow/tools/api/golden/v1/tensorflow.train.-looper-thread.pbtxt [hunk 1]
```diff
@@ -36,7 +36,7 @@ tf_class {
   }
   member_method {
     name: "join"
-    argspec: "args=[\'self\', \'timeout\'], varargs=None, keywords=None, defaults=[\'None\'], "
+    argspec: "args=[\'self\', \'timeout\', \'balancing\'], varargs=None, keywords=None, defaults=[\'None\', \'True\'], "
```
Comment (ezhulenev) [inline tensorflow/tools/api/golden/v1/tensorflow.train.-looper-thread.pbtxt]: That does look suspicious as well

File: tensorflow/tools/api/golden/v1/tensorflow.pbtxt [hunk 2]
```diff
@@ -336,10 +336,6 @@ tf_module {
     name: "errors"
     mtype: "<type \'module\'>"
   }
-  member {
-    name: "estimator"
-    mtype: "<type \'module\'>"
-  }
```
Comment (ezhulenev) [inline tensorflow/tools/api/golden/v1/tensorflow.pbtxt]: What does `estimator` module has to do with MKL quantized ops?

File: tensorflow/tools/api/golden/v2/tensorflow.train.-looper-thread.pbtxt [hunk 3]
```diff
@@ -36,7 +36,7 @@ tf_class {
   }
   member_method {
     name: "join"
-    argspec: "args=[\'self\', \'timeout\'], varargs=None, keywords=None, defaults=[\'None\'], "
+    argspec: "args=[\'self\', \'timeout\', \'balancing\'], varargs=None, keywords=None, defaults=[\'None\', \'True\'], "
```
Comment (ezhulenev) [inline tensorflow/tools/api/golden/v2/tensorflow.train.-looper-thread.pbtxt]: and this

**Comment by ezhulenev** [comment]
Also I have a general concern of adding new `MKL` ops that do not start with `_Mkl` prefix. Was it discussed anywhere? I don't have enough context, but for consistency it seems that all ops that are enabled only with MKL config should start with `_Mkl`.

/cc @martinwicke @tatianashp @penpornk

File: tensorflow/tools/api/golden/v1/tensorflow.train.-looper-thread.pbtxt (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/tools/api/golden/v1/tensorflow.train.-looper-thread.pbtxt]: This was suggested by the api compatibility test in order for the test to pass. I will remove this.

File: tensorflow/tools/api/golden/v1/tensorflow.pbtxt (hunk 2 above)
Comment (mahmoud-abuzaina) [reply tensorflow/tools/api/golden/v1/tensorflow.pbtxt]: same as above, will remove it.

File: tensorflow/tools/api/golden/v2/tensorflow.train.-looper-thread.pbtxt (hunk 3 above)
Comment (mahmoud-abuzaina) [reply tensorflow/tools/api/golden/v2/tensorflow.train.-looper-thread.pbtxt]: will remove it.

**Comment by ezhulenev** [comment]
Is my understanding correct, that you register op specs for Quantized ops (e.g. REGISTER_OP for `QuantizedConv2DAndRelu`), but do not provide any kernel implementation. And only MKL pass through rewrite will make it actually work?

**Review by ezhulenev — DISMISSED** [review]

**Comment by mahmoud-abuzaina** [comment]
Yes, that is correct.

**Comment by asimshankar** [comment]
(For tf-api-review)

Something seems amiss - we don't want to export all these as top level symbols in the Python `tf` namespace, right? (`tf. quantized_conv2d_and_relu`, `tf. quantized_conv2d_and_relu_and_requantize` etc.) 


I'm guessing we do _not_ want to make these operations visible from the Python API but instead hide them in graph rewrites. If so, what you're looking for is probably to add `api_def` files that hide these symbols from the Python API by adding a `visibility: HIDDEN` to `tensorflow/core/api_def/base_api/api_def_<OpName>.pbtxt` similar to https://github.com/tensorflow/tensorflow/blob/30ff2f477908ca4b038b691621d481e842713ecc/tensorflow/core/api_def/base_api/api_def_MaxPoolGrad.pbtxt#L3

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by mahmoud-abuzaina** [comment]
@asimshankar thanks for the explanation. I have marked all Mkl Quantized Ops as HIDDEN. Can you have a look?

**Comment by alextp** [comment]
Since there's no API change here anymore this doesn't need the API review tag.

**Comment by mahmoud-abuzaina** [comment]
@alextp can you review this?

**Review by alextp — APPROVED** [review]

**Comment by mahmoud-abuzaina** [comment]
Thanks @alextp. The failures seem not related to this PR. Can we merge it?

**Pull request merged**